src/core/src/CubicSpline.cpp

```diff
@@ -48,52 +50,61 @@ bool iDynTree::CubicSpline::setData(const iDynTree::VectorDynSize& time, const i
         std::cerr << "[ERROR][CUBICSPLINE] The input data are empty!" << std::endl;
         return false;
     }
-    
+
     if(time.size() != yData.size()){
         std::cerr << "[ERROR][CUBICSPLINE] The input data are expected to have the same dimension: xData = " << time.size() << ", yData = " << yData.size() << "." <<std::endl;
         return false;
     }
-    
+
     if(time.size() < 2){
         std::cerr << "[ERROR][CUBICSPLINE] At least two data points are needed to compute the spline." << std::endl;
         return false;
     }
 
     m_time.resize(time.size());
     m_y.resize(yData.size());
-    
+
     m_coefficients.resize(time.size()-1);
     m_velocities.resize(time.size());
-    
+
     m_T.resize(time.size()-1);
-    
+
     m_time = time;
 
     m_y = yData;
-    
-    return this->computeCoefficients();
+
+    m_areCoefficientsUpdated = false;
+
+    return true;
 }
 
 bool iDynTree::CubicSpline::computeCoefficients()
 {
+    // the coefficients are updated. No need to recompute them
+    if(m_areCoefficientsUpdated)
+        return true;
+
```

Comment:
Could you add braces here just to be safe?

Reply:
Done in ac34f53b1347fb9a81b5922bf06bc1cd741e7672